.github/bin/free-disk-space.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -euo pipefail
+
+function list_installed_packages()
+{
+    apt list --installed "$1" 2>/dev/null | awk -F'/' 'NR>1{print $1}'
+}
+
+function free_up_disk_space_ubuntu()
+{
+    local packages='azure-cli aspnetcore-* dotnet-* ghc-* firefox*
+        google-chrome-* google-cloud-* libmono-* llvm-* imagemagick
+        postgresql-* rubu-* spinxsearch unixodbc-dev mercurial esl-erlang
+        microsoft-edge-stable mono-* msbuild mysql-server-core-*
+        php-* php7* powershell* mongodb* microsoft-edge* subversion ansible'
+
+    for package in $packages; do
```

Comment:
the globs will get expanded here and start matching stuff if a file exists in cwd which matches the pattern. Just convert `packages` to an array.